android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/TranslationsMiddleware.kt

```diff
@@ -146,40 +149,54 @@ class TranslationsMiddleware(
     }
 
     /**
-     * Checks if the translations engine supports the device architecture and updates the state.
-     * When the device does support the translations engine, then the engine will request
-     * initialization of translations data for the [BrowserState.translationEngine] via
-     * [TranslationsAction.InitTranslationsBrowserState].
+     * Use this to initialize translations data for [BrowserState.translationEngine]. If an
+     * issue occurs, the relevant error will be set on [BrowserState.translationEngine].
+     *
+     * This will populate:
+     * Language Support - [requestSupportedLanguages]
      *
      * @param context Context to use to dispatch to the store.
      */
-    private fun requestEngineSupportAndInit(
+    private fun initializeBrowserStore(
```

Comment:
nit: I think we could probably just move the list of function invocations that will happen in this function inline, unless we might call this function from more than one place.

Reply:
I think it is possible we might need to call it for error handling. (We'd probably just recall `InitTranslationsBrowserState`, though as another possibility.) I also have a slight preference to keep it for clarity / self-documenting code too because we are also going to also have a parallel list for `initializeSessionStore`. (Trying to keep the concepts separate and clear as best I can.)